include/kframework/infinite-gas.md

```diff
@@ -5,134 +5,49 @@ Here we use the construct `#gas` to represent positive infinity, while tracking
 This allows (i) computing final gas used, and (ii) never stopping because of out-of-gas.
 Note that the argument to `#gas(_)` is just metadata tracking the current gas usage, and is not meant to be compared to other values.
 As such, any `#gas(G)` and `#gas(G')` are the _same_ positive infinite, regardless of the values `G` and `G'`.
-In particular, this means that `#gas(_) <Int #gas(_) => false`, and `#gas(_) <=Int #gas(_) => true`, regardless of the values contained in the `#gas(_)`.
+In particular, this means that `#gas(_) <Gas #gas(_) => false`, and `#gas(_) <=Gas #gas(_) => true`, regardless of the values contained in the `#gas(_)`.
 
 ```k
 requires "evm.md"
 
 module INFINITE-GAS
-    imports INFINITE-GAS-HASKELL
-endmodule
-
-module INFINITE-GAS-HASKELL [kore]
-    imports INFINITE-GAS-COMMON
-
-    rule #gas(_) <=Int _C => false [concrete(_C), simplification]
-    rule _C <=Int #gas(_) => true  [concrete(_C), simplification]
-
-    rule { #gas(_) #Equals #gas(_) } => #Top [simplification]
-endmodule
-
-module INFINITE-GAS-COMMON
+    imports BASIC-K
+    imports INT
+    imports BOOL
+    imports GAS
     imports EVM
 
- // Generic rules
-
-    rule (notBool (A andBool B)) andBool A => (notBool B) andBool A                      [simplification]
-    rule #if B #then C +Int C1 #else C +Int C2 #fi => C +Int #if B #then C1 #else C2 #fi [simplification]
-
- // Symbolic Gas
-
-    syntax Int ::= #gas ( Int ) [function, total, no-evaluators, klabel(infGas), symbol, smtlib(infGas)]
- // ----------------------------------------------------------------------------------------------------
-    rule #gas(G) +Int G' => #gas(G +Int G') requires 0 <=Int G' orBool 0 -Int G' <Int #gas(G)  [simplification]
-    rule G +Int #gas(G') => #gas(G +Int G') requires 0 <=Int G  orBool 0 -Int G  <Int #gas(G') [simplification]
+    syntax Gas ::= #gas(Int) [klabel(infGas), symbol, smtlib(infGas)]
+
+    rule #gas(G) +Gas G'       => #gas(G +Int G')
+    rule #gas(G) -Gas G'       => #gas(G -Int G')
+    rule #gas(G) *Gas G'       => #gas(G *Int G')
+    rule #gas(G) /Gas G'       => #gas(G /Int G')  requires G' =/=Int 0
+    rule      G  +Gas #gas(G') => #gas(G +Int G')
+    rule      G  -Gas #gas(G') => #gas(G -Int G')
+    rule      G  *Gas #gas(G') => #gas(G *Int G')
+    rule      G  /Gas #gas(G') => #gas(G /Int G')  requires G' =/=Int 0
+    rule #gas(G) +Gas #gas(G') => #gas(G +Int G')
+    rule #gas(G) -Gas #gas(G') => #gas(G -Int G')
+    rule #gas(G) *Gas #gas(G') => #gas(G *Int G')
+    rule #gas(G) /Gas #gas(G') => #gas(G /Int G')  requires G' =/=Int 0
+
+    rule  _:Int    <Gas #gas(_) => true
+    rule  #gas(_)  <Gas _       => false [simplification]
+    rule  #gas(_) <=Gas _:Int   => false
+    rule _        <=Gas #gas(_) => true  [simplification]
```

Comment:
```suggestion
    rule  _:Int    <Gas #gas(_) => true
    rule  #gas(_)  <Gas _       => false
    rule  #gas(_) <=Gas _:Int   => false
    rule _        <=Gas #gas(_) => true
```

Not sure, but it feels weird that these two rules are marked as simplification, and the others are not. Any reason for it?

Reply:
I recall now the reason I had these here was for short circuit evaluation. Maybe I am wrong, but I thought that it would mean that any rules that could apply to the interior of a `#gas(...)` would be skipped. Maybe this is better achieved with priorities?